addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.js

```diff
@@ -48,13 +58,14 @@ export class PartnerDetailsEdit extends Component {
             return false;
         }
     }
-    /**
-     * Save to field `changes` all input changes from the form fields.
-     */
-    captureChange(event) {
-        this.changes[event.target.name] = event.target.value;
-    }
     saveChanges() {
+        if (
+            this.changes.state_id &&
+            this.pos.states.find((state) => state.id == this.changes.state_id).country_id[0] !=
+                this.changes.country_id
+        ) {
+            this.changes.state_id = false;
+        }
```

Comment:
Kindly refactor this such that we use strict (in)equality. I don't want to rely on the use of '==' or '!=' to compare string and int.

Otherwise this PR is good to go. Good job.